server/aesdsocket.c

```diff
@@ -385,12 +394,40 @@ void *socketThreadfunc(void* threadparams)
             syslog(LOG_ERR , "mutex lock\n");
         }
 
-        // Write the data packet to the file
-        int buffer_len = ((datapacket + 1) + (BUFFER_SIZE * realloc_int));
-        int write_len = write(fd , write_buffer , buffer_len);
-        // Error check
-        if(ERROR == write_len){
-            syslog(LOG_ERR , "write\n");
+        // IOCTL logic
+
+        struct aesd_seekto aesd_ioctl;
+
+        // Compare if the string is the same as the required command
+        if (!strncmp(write_buffer , aesd_command , strlen(aesd_command))){
+
+            char *temp_buffer = write_buffer;
+
+            // Extract X
+            temp_buffer += POINTER_FOR_X;
+            aesd_ioctl.write_cmd = (*temp_buffer - '0');
```

Comment:
@KhyatiSatta what does this assume about the number of digits in write_cmd and write_cmd_offset?